src/kernel/kmain.zig

```diff
@@ -83,6 +83,12 @@ export fn kmain(boot_payload: arch.BootPayload) void {
     arch.init(&mem_profile);
     logger.info("Arch init done\n", .{});
 
+    // The VMM runtime tests can't happen until the architecture has initialised itself
+    switch (build_options.test_mode) {
+        .Initialisation => vmm.runtimeTests(arch.VmmPayload, kernel_vmm, &mem_profile),
+        else => {},
+    }
+
```

Comment:
Can't the `arch.init()` be moved up to above the `pmm.init()`?

Reply:
The thing is that the arch initialisation depends on the vmm being set up.